gobblin-service/src/main/java/org/apache/gobblin/service/modules/template_catalog/ObservingFSFlowEdgeTemplateCatalog.java

```diff
@@ -49,6 +51,10 @@
   private Map<URI, List<JobTemplate>> jobTemplateMap = new ConcurrentHashMap<>();
   private ReadWriteLock rwLock;
 
+  @Getter
+  @Setter
+  private boolean shouldRefreshFlowGraph = false;
```

Comment:
See comment earlier about making access of this variable thread safe.

Reply:
Is it necessary to add another lock here? I've added a lock in the flowgraphmonitor (which is where it gets read/set to false), and there is already a lock in this class on when it's set to true.

Reply:
Sorry, I did not mean to add a lock at this assignment. But my comment was more to ensure that access to the variable is guarded by a lock.